src/nbl/asset/utils/CIESProfileParser.cpp

```diff
@@ -167,48 +171,37 @@ bool CIESProfileParser::parse(CIESProfile& result)
     float totalEmissionIntegral = 0.0, nonZeroEmissionDomainSize = 0.0;
     constexpr auto FULL_SOLID_ANGLE = 4.0f * core::PI<float>();
 
-    if (result.symmetry == CIESProfile::ISOTROPIC && result.vAngles.size() == 1) // not sure if any IES provides this kind of distribution but it should be possible to do so meaning fixed candela value for each direction
+    for (size_t i = 0; i < result.hAngles.size() - 2; i++)
     {
-        const auto candelaValue = result.getCandelaValue(0, 0);
-        constexpr auto solidAngle = FULL_SOLID_ANGLE; // spans whole sphere area
+        const float dPhiRad = result.symmetry != CIESProfile::ISOTROPIC ? (hAngles[i + 1] - hAngles[i]) : core::PI<float>() * 2.0f;
 
-        totalEmissionIntegral = candelaValue * solidAngle;
-        nonZeroEmissionDomainSize = solidAngle;
-    }
-    else
-        for (size_t i = 0; i < result.hAngles.size(); i++)
-            for (size_t j = 0; j < result.vAngles.size(); j++)
+        for (size_t j = 0; j < result.vAngles.size() - 2; j++)
+        {
+            const auto candelaValue = result.getCandelaValue(i, j);
+
+            if (result.maxCandelaValue < candelaValue)
+                result.maxCandelaValue = candelaValue;
+
+            const float thetaRad = core::radians<float>(result.vAngles[j]);
+            const float dThetaRad = core::radians<float>((j < result.vAngles.size() - 1) ? result.vAngles[j + 1] - result.vAngles[j] : result.vAngles[j] - result.vAngles[j - 1]);
+
+            const auto differentialSolidAngle = std::sin(thetaRad) * dThetaRad * dPhiRad;
```

Comment:
remember about proper integration of the `sin` as a `cos`

Reply:
> remember about proper integration of the `sin` as a `cos`

we use [physics convention](https://en.wikipedia.org/wiki/Spherical_coordinate_system#/media/File:3D_Spherical.svg) for spherical coordinates hence differential solid angle is 

$$
{\displaystyle \mathrm {d} \Omega ={\frac {\mathrm {d} S_{r}}{r^{2}}}=\sin\theta*\mathrm {d} \theta *\mathrm {d} \varphi .}
$$

however if we were to swap polar with azimuth (mathematics convention) then we would use the `cos` end up with

$$
{\displaystyle \mathrm {d} \Omega ={\frac {\mathrm {d} S_{r}}{r^{2}}}=\cos\theta*\mathrm {d} \theta *\mathrm {d} \varphi .}
$$

Reply:
ah shit but 0 degrees is Z- IES so it makes sense to turn `sin` into `cos` as the part of this integration, okay okay

Reply:
nooo, this is not what I meant

Reply:
> > remember about proper integration of the `sin` as a `cos`
> 
> we use [physics convention](https://en.wikipedia.org/wiki/Spherical_coordinate_system#/media/File:3D_Spherical.svg) for spherical coordinates hence differential solid angle is
> 
> dΩ=dSrr2=sin⁡θ∗dθ∗dφ.
> 
> however if we were to swap polar with azimuth (mathematics convention) then we would use the `cos` & end up with
> 
> dΩ=dSrr2=cos⁡θ∗dθ∗dφ.

![image](https://github.com/Devsh-Graphics-Programming/Nabla/assets/6894321/c8781ff6-ad8e-4e41-b7e1-aa35493dea49)

This is why it should be cos, not because of anything else.

alos tehta=0 being Z- or Z+ changes nothing, its either sin(tehta) or sin(pi-theta)

Reply:
you're not doing Monte Carlo or some other sampling thing, this is pure finite element stuff

Reply:
resolved in following comment

Reply:
> ah one more thing, the `candelaValue` you multiply the solid angle with should be an average of `i,j` , `i+1,j`, `i,j+1`, and `i+1,j+1`

for non isotropic cases this interpolation is good, handled also isotropic case